fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushdownFilterThroughSort.java

```diff
@@ -19,24 +19,36 @@
 
 import org.apache.doris.nereids.rules.Rule;
 import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.Slot;
 import org.apache.doris.nereids.trees.plans.Plan;
 import org.apache.doris.nereids.trees.plans.logical.LogicalFilter;
 import org.apache.doris.nereids.trees.plans.logical.LogicalSort;
+import org.apache.doris.nereids.util.ExpressionUtils;
+
+import java.util.Set;
 
 /**
  * Push down filter through sort.
  * input:
  * filter -> sort
  * output:
- * sort -> filter
+ * [sort] -> filter
  * The filter can be directly push down to the sort.
+ * Note when the sort key is equal to Literal, the sort can be eliminated
  */
 public class PushdownFilterThroughSort extends OneRewriteRuleFactory {
     @Override
     public Rule build() {
         return logicalFilter(logicalSort()).then(filter -> {
             LogicalSort<Plan> sort = filter.child();
+            if (checkSlotsConstant(sort.getInputSlots(), filter)) {
+                return new LogicalFilter<>(filter.getConjuncts(), sort.child());
```

Comment:
could we expand this rule to other sceario, such as group by constants, and uncorelate subuquery?